src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForFailedStatus.java

```diff
@@ -228,20 +229,31 @@ public void should_record_signature_failure_when_zip_contains_invalid_signature(
 
         // then
         envelopeWasNotCreated();
-        eventWasCreated(Event.DOC_SIGNATURE_FAILURE);
+        eventsCreated(Event.DOC_SIGNATURE_FAILURE);
         errorWasSent(SAMPLE_ZIP_FILE_NAME, ErrorCode.ERR_SIG_VERIFY_FAILED);
     }
 
-    private void eventWasCreated(Event event) {
+    private void eventsCreated(Event event) {
         List<ProcessEvent> processEvents = processEventRepository.findAll();
-        assertThat(processEvents).hasSize(1);
 
-        ProcessEvent processEvent = processEvents.get(0);
-
-        assertThat(processEvent.getContainer()).isEqualTo(testContainer.getName());
-        assertThat(processEvent.getEvent()).isEqualTo(event);
-        assertThat(processEvent.getId()).isNotNull();
-        assertThat(processEvent.getReason()).isNotBlank();
+        // Verify that Zip file processing event is created
+        assertThat(processEvents)
+            .filteredOn(processEvent -> processEvent.getEvent().equals(Event.ZIPFILE_PROCESSING_STARTED))
+            .hasOnlyOneElementSatisfying(processEvent -> {
+                assertThat(processEvent.getContainer()).isEqualTo(testContainer.getName());
+                assertThat(processEvent.getEvent()).isEqualTo(ZIPFILE_PROCESSING_STARTED);
+                assertThat(processEvent.getId()).isNotNull();
+                assertThat(processEvent.getReason()).isNull();
+            });
+
+        assertThat(processEvents)
+            .filteredOn(processEvent -> processEvent.getEvent().equals(event))
+            .hasOnlyOneElementSatisfying(processEvent -> {
+                assertThat(processEvent.getContainer()).isEqualTo(testContainer.getName());
+                assertThat(processEvent.getEvent()).isEqualTo(event);
+                assertThat(processEvent.getId()).isNotNull();
+                assertThat(processEvent.getReason()).isNotBlank();
+            });
```

Comment:
Please suggest if there is a better way to verify these events

Reply:
checking for `ZIPFILE_PROCESSING_STARTED` is hidden and not obvious, as code says `
```
eventsCreated(DOC_SIGNATURE_FAILURE)
```

Maybe it would be clearer to have one of those two options instead:
```
eventWasCreated(Event.DOC_SIGNATURE_FAILURE);
eventWasCreated(Event.ZIPFILE_PROCESSING_STARTED);
```
or
```
eventsWereCreated(
    ZIPFILE_PROCESSING_STARTED,
    DOC_SIGNATURE_FAILURE
);
```

Reply:
or just do what's in tests below
```
        assertThat(processEventRepository.findAll())
            .hasSize(2)
            .hasSize(3)
            .extracting(e -> tuple(e.getContainer(), e.getZipFileName(), e.getEvent(), e.getReason()))
            .containsOnly(
                tuple(testContainer.getName(), zipFileName, ZIPFILE_PROCESSING_STARTED, null),
                tuple(testContainer.getName(), zipFileName, DOC_UPLOAD_FAILURE, failureReason),
                tuple(testContainer.getName(), zipFileName, DOC_UPLOAD_FAILURE, "oh no")
            );
```

Reply:
@rkondratowicz  thank you! I went with the last suggestion